### PR TITLE
{numlib}[foss/2020a,intel/2020a] SuperLU_DIST v6.4.0

### PR DESCRIPTION
**Pull request by Darkless012**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 3 out of 4 (2 easyconfigs in total)
vera-c1 - Linux centos linux 7.8.2003, x86_64, Intel Xeon Processor (Skylake), Python 2.7.5
See https://gist.github.com/084d97ddf1c94dd1e1a9dade1392fff2 for a full test report.

**Comment by Darkless012** [comment]
I got randomly those errors as well. Not sure what causes that.
(extracted from the report above)
```
      Start  1: pdtest_1x1_1_2_8_20_SP
 1/24 Test  #1: pdtest_1x1_1_2_8_20_SP ...........***Failed    1.69 sec
Time to read and distribute matrix 0.00
[vera-c1:3389 :0:3418] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x4d)
==== backtrace (tid:   3418) ====
 0 0x00000000000214ae ucs_debug_print_backtrace()  /local/EB/build/UCX/1.8.0/GCCcore-9.3.0/ucx-1.8.0/src/ucs/debug/debug.c:653
 1 0x000000000034a552 mkl_blas_avx512_xdgemv()  ???:0
 2 0x000000000022bb4e mkl_blas_xdgemv()  ???:0
 3 0x0000000000250b6a mkl_blas_dgemv()  ???:0
 4 0x00000000002fba86 mkl_blas_dgemm()  ???:0
 5 0x000000000019251f DGEMM()  ???:0
 6 0x0000000000476fea dlsum_bmod_inv()  ???:0
 7 0x0000000000476ee1 dlsum_bmod_inv()  ???:0
 8 0x00000000000e12c2 _INTERNALfadf56ac::__kmp_invoke_task()  /nfs/site/proj/openmp/promo/20200205/tmp/lin_32e-rtl_int_5_nor_dyn.rel.c0.s0.t1..h1.w1-fxe16lin03/../../src/kmp_tasking.cpp:1782
 9 0x00000000000ecdcc _INTERNALfadf56ac::__kmp_execute_tasks_template<kmp_flag_64<false, true> >()  /nfs/site/proj/openmp/promo/20200205/tmp/lin_32e-rtl_int_5_nor_dyn.rel.c0.s0.t1..h1.w1-fxe16lin03/../../src/kmp_tasking.cpp:3185
10 0x00000000000ecdcc __kmp_execute_tasks_64<false, true>()  /nfs/site/proj/openmp/promo/20200205/tmp/lin_32e-rtl_int_5_nor_dyn.rel.c0.s0.t1..h1.w1-fxe16lin03/../../src/kmp_tasking.cpp:3284
11 0x000000000006bb01 kmp_flag_64<false, true>::execute_tasks()  /nfs/site/proj/openmp/promo/20200205/tmp/lin_32e-rtl_int_5_nor_dyn.rel.c0.s0.t1..h1.w1-fxe16lin03/../../src/kmp_wait_release.h:964
12 0x000000000006bb01 _INTERNAL51694e09::__kmp_wait_template<kmp_flag_64<false, true>, true, false, true>()  /nfs/site/proj/openmp/promo/20200205/tmp/lin_32e-rtl_int_5_nor_dyn.rel.c0.s0.t1..h1.w1-fxe16lin03/../../src/kmp_wait_release.h:374
13 0x000000000006e23d kmp_flag_64<false, true>::wait()  /nfs/site/proj/openmp/promo/20200205/tmp/lin_32e-rtl_int_5_nor_dyn.rel.c0.s0.t1..h1.w1-fxe16lin03/../../src/kmp_wait_release.h:971
14 0x000000000007526b __kmp_fork_barrier()  /nfs/site/proj/openmp/promo/20200205/tmp/lin_32e-rtl_int_5_nor_dyn.rel.c0.s0.t1..h1.w1-fxe16lin03/../../src/kmp_barrier.cpp:2369
15 0x00000000000b1170 __kmp_launch_thread()  /nfs/site/proj/openmp/promo/20200205/tmp/lin_32e-rtl_int_5_nor_dyn.rel.c0.s0.t1..h1.w1-fxe16lin03/../../src/kmp_runtime.cpp:6080
16 0x000000000012d19c _INTERNAL27dd4e00::__kmp_launch_worker()  /nfs/site/proj/openmp/promo/20200205/tmp/lin_32e-rtl_int_5_nor_dyn.rel.c0.s0.t1..h1.w1-fxe16lin03/../../src/z_Linux_util.cpp:593
17 0x0000000000007ea5 start_thread()  pthread_create.c:0
18 0x00000000000fe8dd __clone()  ???:0
=================================
```

**Comment by boegel** [comment]
Could be a bug in Intel MKL triggered by the SuperLU tests...

File: easybuild/easyconfigs/s/SuperLU_DIST/SuperLU_DIST-6.4.0-foss-2020a.eb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ('ParMETIS', '4.0.3'),
+]
+
+configopts = '-DTPL_PARMETIS_INCLUDE_DIRS="${EBROOTPARMETIS}/include" '
+configopts += '-DTPL_PARMETIS_LIBRARIES="${EBROOTPARMETIS}/lib/libparmetis.a;${EBROOTPARMETIS}/lib/libmetis.a" '
+
+# Some tests run longer than default 1500s timeout on fairly big machine (36 cores).
+# Increasing timeout to 3000 resolves the timeout error.
+# Be ware that tests run ~2hrs.
+pretestopts = 'export ARGS="$ARGS --timeout 3000" && '
```
Comment (boegel) [inline easybuild/easyconfigs/s/SuperLU_DIST/SuperLU_DIST-6.4.0-foss-2020a.eb:30]: I'd say this is too excessive to run during the installation...

Is there any way to only run a subsuite of the tests?
If not, we should leave this commented out, imagine doing the installation on just 3 cores...

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/SuperLU_DIST/SuperLU_DIST-5.3.0-intel-2020a.eb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+easyblock = "EB_SuperLU"
+
+name = 'SuperLU_DIST'
+version = '5.3.0'
```
Comment (boegel) [inline easybuild/easyconfigs/s/SuperLU_DIST/SuperLU_DIST-5.3.0-intel-2020a.eb:4]: There's a more recent 5.3.0 variant available tagged as `5.3.0-trisolve`, maybe using that one is better?

See https://github.com/xiaoyeli/superlu_dist/compare/v5.3.0...v5.3.0-trisolve

BTW, I think it makes sense to throw this easyconfig (and the change in the test suite) in a separate PR...

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/SuperLU_DIST/SuperLU_DIST-5.3.0-intel-2020a.eb (hunk 2 above)
Comment (Darkless012) [reply easybuild/easyconfigs/s/SuperLU_DIST/SuperLU_DIST-5.3.0-intel-2020a.eb]: Ye, it makes sense, will create another PR.

File: easybuild/easyconfigs/s/SuperLU_DIST/SuperLU_DIST-6.4.0-foss-2020a.eb (hunk 1 above)
Comment (Darkless012) [reply easybuild/easyconfigs/s/SuperLU_DIST/SuperLU_DIST-6.4.0-foss-2020a.eb:30]: Is Patchfile with modifications to Ctest a way?

**Comment by Darkless012** [comment]
This is only foss/2020a version, which passed tests. Please retest. (should be possible to merge)
intel2020a version will be split to another PR.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/s/SuperLU_DIST/SuperLU_DIST-6.4.0-foss-2020a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/s/SuperLU_DIST/SuperLU_DIST-6.4.0-foss-2020a.eb:30]: Maybe we can use the `--tests-regex` option to pick a couple of tests, or use `--exclude-regex` to exclude the very long test(s)?

**Comment by Darkless012** [comment]
Rerun tests please. included tests now are minimal only.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3501.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/9f0cce6389452000769bb74e8cdcb5fb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2676.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/70123989f5ab8c43b76b5b379b51aafc for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3100.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/699d818b3741129094afedfcf5bbc730 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @Darkless012!

**Pull request merged**